tests/raft_cluster_integration.rs

```diff
@@ -2932,3 +2932,444 @@ async fn test_progressive_node_failure_and_recovery() {
 
     eprintln!("\n=== test_progressive_node_failure_and_recovery PASSED ===");
 }
+
+/// Test snapshot transfer between nodes
+///
+/// This test verifies that when a lagging node rejoins the cluster,
+/// it correctly receives and applies a snapshot from the leader.
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn test_snapshot_transfer_between_nodes() {
```

Comment:
I dont see how this test is asserting that the catch up happened via snapshot as opposed to log replay.